core/arch/arm/kernel/link.mk

```diff
@@ -131,12 +131,20 @@ endef
 $(link-out-dir)/version.o:
 	$(call gen-version-o)
 
+-include $(link-out-dir)/.tee.elf.cmd
+define check-link-objs
+$(if $(strip $(filter-out $(link-objs), $(old-link-objs))
+	     $(filter-out $(old-link-objs), $(link-objs))), FORCE_LINK := FORCE)
+endef
+$(eval $(call check-link-objs))
```

Comment:
If a TA is modified, the C file will become outdated and will be generated again. It's not something that is handled by this. This only takes care of the situation when no files are modified (because they are already all present and up-to-date from a previous build), but tee.elf should still be re-linked to add/remove files.